minio/api.py

```diff
@@ -300,9 +300,9 @@ def list_buckets(self):
             raise ResponseError(response, method).get_exception()
         try:
             return parse_list_buckets(response.data)
-        except InvalidXMLError:
-            if self._endpoint_url.endswith("s3.amazonaws.com"):
-                raise InvalidArgumentError('Invalid access_key and secret_key.')
+        except InvalidXMLError as e:
+            if self._endpoint_url.endswith("s3.amazonaws.com") and (not self._access_key or not self._secret_key):
+                raise InvalidAccessKeyId(e)
```

Comment:
Actually AccessDenied is better @poornas - since we need to indicate that signature is needed and anonymous request is not allowed.  We shouldn't state access key is wrong - that would mean that server validated access key to be wrong.